ZelBack/src/services/appsService.js

```diff
@@ -6056,30 +6069,71 @@ async function storeAppRunningMessage(message) {
-    name: message.name,
-    hash: message.hash, // hash of application specifics that are running
-    ip: message.ip,
-    broadcastedAt: new Date(message.broadcastedAt),
-    expireAt: new Date(validTill),
-  };
 
-  // indexes over name, hash, ip. Then name + ip and name + ip + broadcastedAt.
-  const queryFind = { name: newAppRunningMessage.name, ip: newAppRunningMessage.ip, broadcastedAt: { $gte: newAppRunningMessage.broadcastedAt } };
-  const projection = { _id: 0 };
-  // we already have the exact same data
-  const result = await dbHelper.findOneInDatabase(database, globalAppsLocations, queryFind, projection);
-  if (result) {
-    // it is already stored
-    return false;
+  if (message.version === 1) {
+    const newAppRunningMessage = {
+      name: message.name,
+      hash: message.hash, // hash of application specifics that are running
+      ip: message.ip,
+      broadcastedAt: new Date(message.broadcastedAt),
+      expireAt: new Date(validTill),
+    };
+
+    // indexes over name, hash, ip. Then name + ip and name + ip + broadcastedAt.
+    const queryFind = { name: newAppRunningMessage.name, ip: newAppRunningMessage.ip, broadcastedAt: { $gte: newAppRunningMessage.broadcastedAt } };
+    const projection = { _id: 0 };
+    // we already have the exact same data
+    const result = await dbHelper.findOneInDatabase(database, globalAppsLocations, queryFind, projection);
+    if (result) {
+      // it is already stored
+      return false;
+    }
+    const queryUpdate = { name: newAppRunningMessage.name, ip: newAppRunningMessage.ip };
+    const update = { $set: newAppRunningMessage };
+    const options = {
+      upsert: true,
+    };
+    await dbHelper.updateOneInDatabase(database, globalAppsLocations, queryUpdate, update, options);
+  } else {
+    let messageNotOk = false;
+    for (let i = 0; i < message.apps.length; i += 1) {
+      const app = message.apps[i];
+      if (typeof app.hash !== 'string' || typeof app.name !== 'string') {
+        return new Error('Invalid Flux App Running v2 message for storing');
+      }
+      const newAppRunningMessage = {
+        name: app.name,
+        hash: app.hash, // hash of application specifics that are running
+        ip: message.ip,
+        broadcastedAt: new Date(message.broadcastedAt),
+        expireAt: new Date(validTill),
+      };
+
+      // indexes over name, hash, ip. Then name + ip and name + ip + broadcastedAt.
+      const queryFind = { name: newAppRunningMessage.name, ip: newAppRunningMessage.ip, broadcastedAt: { $gte: newAppRunningMessage.broadcastedAt } };
+      const projection = { _id: 0 };
+      // we already have the exact same data
+      // eslint-disable-next-line no-await-in-loop
+      const result = await dbHelper.findOneInDatabase(database, globalAppsLocations, queryFind, projection);
+      if (result) {
+        // found a message that was already stored/bad message
+        messageNotOk = true;
+        break;
```

Comment:
Here we should do continue to process the other parts of the message, apps that can be running

Reply:
Actually OK. Lets just break it and not process further.